lib/streamlit/runtime/state/query_params_proxy.py

```diff
@@ -0,0 +1,86 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, Dict, Iterator, List, Union
+
+from streamlit.runtime.state.query_params import (
+    QueryParams,
+    _missing_key_error_message_query_params,
+)
+from streamlit.runtime.state.session_state_proxy import get_session_state
+from streamlit.type_util import Key
+
+
+def get_query_params() -> QueryParams:
+    from streamlit.runtime.scriptrunner import get_script_run_ctx
+
+    ctx = get_script_run_ctx()
+
+    if ctx is None:
+        return QueryParams()
+    return get_session_state()._state._query_params
```

Comment:
I'm pretty sure there are some thread safety issues here since the object returned by `get_session_state` is a `SafeSessionState`, which normally protects access to the internal `_state` object in its methods with a lock, and this is being bypassed here.

Reply:
Thanks for this. I think we just need a method that locks when we are getting the query params from my perspective right?:
```
def get_query_params(self) -> QueryParams:
      with self._lock:
          return self._state._query_params
```